src/shipit_static_analysis/shipit_static_analysis/workflow.py

```diff
@@ -161,8 +167,78 @@ def run(self, revision):
             # Run analyzer on version and store generated issues
             issues += analyzer.run(revision)
 
+        logger.info('Detected {} issue(s)'.format(len(issues)))
+        if not issues:
+            logger.info('No issues, stopping there.')
+            return
+
+        # Build a potential improvement patch
+        self.build_improvement_patch(revision, issues)
+
         # Publish reports about these issues
-        return
         with stats.api.timer('runtime.reports'):
             for reporter in self.reporters.values():
                 reporter.publish(issues, revision)
+
+    def build_improvement_patch(self, revision, issues):
+        '''
+        Build a Diff to improve this revision (styling from clang-format)
+        '''
+        assert isinstance(issues, list)
+
+        # Only use publishable issues
+        # and sort them by filename
+        issues = sorted(
+            filter(lambda i: i.is_publishable(), issues),
+            key=lambda i: i.path,
+        )
+
+        # Apply a patch on each modified file
+        for filename, file_issues in itertools.groupby(issues, lambda i: i.path):
+            full_path = os.path.join(self.repo_dir, filename)
+            assert os.path.exists(full_path), \
+                'Modified file not found {}'.format(full_path)
+
+            # Build raw "ed" patch
+            patch = '\n'.join(issue.as_diff() for issue in file_issues)
+            if not patch:
+                continue
+
+            # Write patch in tmp
+            _, patch_path = tempfile.mkstemp(suffix='.diff')
+            with open(patch_path, 'w') as f:
+                f.write(patch)
+
+            # Apply patch on repository file
```

Comment:
I assume this is to check that the patch is well formatted? Can you add a comment explaining it? Or maybe a message in the assertion `assert exit.returncode == 0`